### PR TITLE
netbox 3.6 support

### PR DESCRIPTION
**Pull request by ryanmerolle**
# Pull Request

## Related Issue

#159 

## New Behavior

Development:

- [x] devcontainer bumped to NetBox 3.6
- [x] migrate from vscode user given ubtuntu uid conflicts

NetBox Changes to investigate

- [ ] The device_role field on the Device model has been renamed to role. The device_role field has been temporarily retained on the REST API serializer for devices for backward compatibility, but is read-only.
- [ ] The choices array field has been removed from the CustomField model. Any defined choices are automatically migrated to CustomFieldChoiceSets, accessible via the new choice_set field on the CustomField model.
- [ ] netbox-community/netbox#12988 -Custom Field Choice Sets
- [ ] netbox-community/netbox#11766  - Remove obsolete custom ChoiceField and MultipleChoiceField classes
- [ ] The device_role field on the Device model has been renamed to role. The device_role field has been temporarily retained on the REST API serializer for devices for backward compatibility, but is read-only.

## Contrast to Current Behavior

...

## Discussion: Benefits and Drawbacks

...

## Changes to the Documentation

...

## Proposed Release Note Entry

...

## Double Check

* [ ] I have explained my PR according to the information in the comments
 or in a linked issue.
* [ ] My PR targets the `dev` branch.

### CODE REVIEW

**Comment by ryanmerolle** [comment]
@Cormoran96 @abhi1693 @jsenecal tagging for visibility.

I started looking at change for 3.6.  Development in VS Code devcontainer now works, but all views except for ACL Interface Assignments are broken.

I'm not currently leveraging NetBox at my new job, but that will likely eventually change.  Just explaining why this work is not already done.

**Comment by garnser** [comment]
Tested, works for 3.7 as well

File: .devcontainer/Dockerfile-plugin_dev [hunk 1]
```diff
@@ -1,28 +1,27 @@
...
-ARG NETBOX_INITIALIZERS_VARIANT=3.5.*
+ARG NETBOX_INITIALIZERS_VARIANT=3.6.*
 
 ARG DEBIAN_FRONTEND=noninteractive
 
 # Install APT packages
 # hadolint ignore=DL3008
 RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
-    && apt-get -y install --no-install-recommends  curl git make openssh-client python3.10-dev sudo wget zsh \
+    && apt-get -y install --no-install-recommends  curl git make openssh-client python3.11-dev sudo wget zsh \
```
Comment (abhi1693) [inline .devcontainer/Dockerfile-plugin_dev:12]:
```suggestion
    && apt-get -y install --no-install-recommends  curl git make openssh-client python3.12-dev sudo wget zsh \
```

**Review by abhi1693 — COMMENTED** [review]
If my suggestion is not required, let's release these changes

**Comment by jeffgdotorg** [comment]
@abhi1693 @ryanmerolle between this PR and @kbelokon's draft PR #175, I think we have all the pieces to get this plugin working through NetBox 3.7. How do we break the logjam?

Let me know if you need me to remove any obstacles. The two hotspots identified by Sonar look okay to me; if you concur, and if the checks all pass after marking them safe, I'm a +1 to merging once the 3.7 changes are in.

PS I tested the fix branch for #175 in a 3.7.4 system and it did not blow up in my face.

**Pull request merged**